UTC-330: Add image hover component.

UTCbridget opened the pull request:
UTC-330: Add image hover component. This component is responsive, is available for both horizontal and vertical images and is built based on cards with much more limited capability.

**Desktop version:**
![sample-hover-effects](https://user-images.githubusercontent.com/82905787/134690861-4c282b56-70c6-4819-979d-9e988e3129f0.gif)

**Mobile version:** 
NOTE: On the desktop version, the image itself is a link accessed through a hover (shows text) then a click (go to the next page). On the mobile version, an actual button was needed because there is no "hover" with a finger action. Therefore, a tap on the image shows the text. The button shows to provide a means to navigating to the next page.
![sample-hover-effects-mobile](https://user-images.githubusercontent.com/82905787/134690377-cc7059c6-fda6-4c46-86af-33cb03a992fa.gif)

**How to implement:**
![sample-hover-effects-admin](https://user-images.githubusercontent.com/82905787/134690406-dd34c54b-a822-437b-9ce8-0a8a91fb0ab2.gif)

UTCbridget committed:
UTC-330: Add image hover component.

UTC-330: Add image hover component.

UTCbridget committed:
UTC-330: Add block extension for editing.

Update block--utc-image-hover-effect.html.twig with block.html.twig extension in order to be able to edit this block in layout builder.

bmartinez287 commented on source/default/_patterns/00-protons/legacy/css/components/UTC-custom-blocks/_utc_hover_images.css:
```diff
@@ -0,0 +1,133 @@
+
+figure.utc-hover-image-effect img {
+  @apply max-w-full opacity-100 w-full;
+  -webkit-transition: opacity 0.35s;
```
that's interesting, I rarely see webkit calls on vanilla css. Was this compiled from less or SASS into css? I think postcss can add the vendor prefixes if we used the native css class `transition: opacity .25s`.

UTCbridget replied:
Ok. Old school habit, I guess.

bmartinez287 commented:
Looks good what would we called it though? If we need to make a distinction between general cards and this one. What would this one we called to avoid confusion or people using it for the wrong purpose.

bmartinez287 commented on apps/drupal-default/particle_theme/templates/block/block--utc-image-hover-effect.html.twig:
```diff
@@ -0,0 +1,106 @@
+	{% set hover_count = content["#block_content"].field_hover_image_count.0.value %}
+	{% set hover_orientation = content["#block_content"].field_hover_image_orientation.0.value %}
+	
+	{% if hover_count == 1 %}
```
what if count == 0? are we enforcing 1 as the default?

UTCbridget replied:
I did think of making all the "one" fields required. Picking "none" shouldn't do anything so the person applying that would hopefully figure that out. If I implement reuseable fields for each image, then I can't make the fields required.

bmartinez287 commented on apps/drupal-default/particle_theme/templates/block/block--utc-image-hover-effect.html.twig:
```diff
@@ -0,0 +1,106 @@
+	{# set variables for card content based on Drupal input #}
+
+	{% set item_hover_1 = {
+		hover_image: content.field_hover_image_one|field_value,
```
I think we might be able to cut down in the number of fields by doing what we did on the button group.
So there would be only one field_hover_image instead of field_hover_image_one field_hover_image_two and field_hover_image_three and we can access the values trough a for loop.

Basically we create a field_hover_image and then the second time we want to add a field we reuse that field which is an option right next to create.

bmartinez287 replied:
https://github.com/UTCWeb/particle/blob/develop/apps/drupal-default/particle_theme/templates/block/block--block-content--utc-button-group.html.twig

UTCbridget replied:
I will look into this.

brittrich replied:
@bmartinez287 is there a way to make entire field groups optionally able to be duplicated? For example, keeping the title, description, link, etc. together as a set and duplicate the entire set of fields each time as needed to add cards (or by the numbers items). Just thinking, if the image field can be duplicated and then the title field can be duplicated...but they aren't grouped logically...it might cause some user confusion when building?

Also, is this module relevant when trying to do something like this? https://www.drupal.org/project/business_rules

bmartinez287 replied:
@UTCbridget I forgot to reply. We are currently researching and prototyping with a couple of alternatives to simplify the twig templates. Bridget is looking into paragraph components and I'm looking into extending fields through custom modules. We should have something by tomorrow night that could simplify the UI on these components. This is currently the main priority we are working on as we want to get all your components in before November.

bmartinez287 replied:
@brittrich if you could prioritize improving the Figma files that be great this week. Mostly since new components will change slightly once we provide a new way for them to be grouped

brittrich replied:
Sounds great and makes perfect sense! I'll spend my time this week in Figma.

UTCbridget committed:
UTC-330: Fix tailwind.config.js

Stray comma removed.

UTCbridget committed:
UTC-330: Fix tailwind.config.js file.

This one doesn't show, but there was code from another pull request interfering with this one. This is a restoration of original files with a comma fix.

UTCbridget committed:
UTC-330: Remove dist files from the PR.

UTC-330: Remove dist files from the PR.

UTCbridget committed:
UTC-330: Remove two more dist files from PR.

UTC-330: Remove two more dist files from PR.

UTCbridget committed:
UTC-330: Fix tailwind.config.js PR.

UTC-330: Fix tailwind.config.js PR picked up from another PR.

UTCbridget committed:
UTC-330:  Fix tailwind.config.js coupled with another PR change.

UTC-330:  My first PR incorrectly mixed my changes with another developer's changes in the tailwind.config.js. This commit includes both mine and the other developer's changes (utchighlight).

brittrich commented:
I LOVE this component, Bridget!
